src/nvim/api/private/helpers.c

```diff
@@ -644,10 +644,8 @@ static Object vim_to_object_rec(typval_T *obj, PMap(ptr_t) *lookup)
 
   switch (obj->v_type) {
     case VAR_STRING:
-      if (obj->vval.v_string != NULL) {
-        rv.type = kObjectTypeString;
-        rv.data.string = cstr_to_string((char *) obj->vval.v_string);
-      }
+      rv.type = kObjectTypeString;
+      rv.data.string = cstr_to_string((char *) obj->vval.v_string);
```

Comment:
The orignal code came from https://github.com/neovim/neovim/commit/ba11128077845d286d9f1a8282e1404d2c80b2a5 which was adapted from if_py. I remember noticing this decision, but I don't know the reason. In most languages like java/c#/etc., a string can be null...

In fact I remember now that this is come up before, and the solution was to fix the call site that was sending a null. Can that be done in this case too?